setting up docker images for malaria24

sewagodimo opened the pull request.

sewagodimo committed:
setting up docker images for malaria24

miltontony commented on Dockerfile:
```diff
@@ -0,0 +1,11 @@
+# https://github.com/praekeltfoundation/docker-seed
+FROM praekeltfoundation/django-bootstrap:py2.7-jessie
+
+ COPY . /app
```
spacing before `COPY`
```suggestion
COPY . /app
```

miltontony commented on Dockerfile:
```diff
@@ -0,0 +1,11 @@
+RUN pip install -e git+git://github.com/onaio/onapie.git#egg=onapie &&\
+    pip install -e .
+
+ ENV DJANGO_SETTINGS_MODULE "malaria24.settings"
```
spacing
```suggestion
ENV DJANGO_SETTINGS_MODULE "malaria24.settings"
```

miltontony commented on Dockerfile:
```diff
@@ -0,0 +1,11 @@
+RUN pip install -e git+git://github.com/onaio/onapie.git#egg=onapie &&\
+    pip install -e .
+
+ ENV DJANGO_SETTINGS_MODULE "malaria24.settings"
```
We also typically use a specific settings file that has docker specific settings. See [molo-gem example](https://github.com/praekeltfoundation/molo-gem/blob/develop/Dockerfile#L4).

miltontony replied:
And the related settings file looks like [this](https://github.com/praekeltfoundation/molo-gem/blob/develop/gem/settings/docker.py)

sewagodimo replied:
I will add the settings file

sewagodimo committed:
remove spaces

sewagodimo committed:
add a dockerignore file

sewagodimo committed:
add docker settings

miltontony commented on Dockerfile:
```diff
@@ -0,0 +1,11 @@
+FROM praekeltfoundation/django-bootstrap:py2.7-jessie
+
+COPY . /app
+RUN pip install -e git+git://github.com/onaio/onapie.git#egg=onapie &&\
```
```suggestion
RUN pip install git+git://github.com/onaio/onapie.git#egg=onapie &&\
```

JayH5 replied:
If you use git to install packages then you also need to install git (`RUN apt-get-install.sh git`)

sewagodimo committed:
fix requirements and onapie install command

sewagodimo committed:
add the pdf packages

sewagodimo committed:
clean up the docker file

sewagodimo committed:
add the seaworthy job to travis

sewagodimo committed:
add seaworthy dockertests

sewagodimo committed:
add tools for the pytest setup

sewagodimo committed:
rename the seaworthy tests

pytest was picking them up

miltontony commented on Dockerfile:
```diff
@@ -0,0 +1,17 @@
+ENV DJANGO_SETTINGS_MODULE "malaria24.settings"
+RUN python manage.py collectstatic --noinput
+
+RUN apt-get-install.sh wkhtmltopdf xvfb
```
This should typically be directly after the `FROM`

sewagodimo replied:
this one RUN or both of them, that is the line after

miltontony replied:
Just this line

miltontony commented on Dockerfile:
```diff
@@ -0,0 +1,17 @@
+RUN pip install https://github.com/onaio/onapie/archive/develop.zip#egg=onapie &&\
+    pip install -e .
+
+ENV DJANGO_SETTINGS_MODULE "malaria24.settings"
```
```suggestion
ENV DJANGO_SETTINGS_MODULE "malaria24.settings.docker"
```

miltontony commented on malaria24/settings/docker.py:
```diff
@@ -0,0 +1,35 @@
+
+DATABASES = {
+    'default': dj_database_url.config(
+        default='sqlite:///%s' % (join(PROJECT_ROOT, 'gemmolo.db'),))}
```
```suggestion
        default='sqlite:///%s' % (join(PROJECT_ROOT, 'malaria24.db'),))}
```

miltontony committed:
Update Dockerfile

Co-Authored-By: sewagodimo <sewagodimo.matlapeng@gmail.com>

sewagodimo committed:
move the run command upwards

sewagodimo committed:
fixed db name

sewagodimo committed:
add compress to dockerfile

sewagodimo committed:
add a secret key for the docker-settings

sewagodimo committed:
remove printline from tests

sewagodimo committed:
add default secret key to base-settings

sewagodimo committed:
add compress command to travis

sewagodimo committed:
add settings

sewagodimo committed:
add --force to the compress file

sewagodimo committed:
fix some settings issues

sewagodimo committed:
add a vumi key to the settings

sewagodimo committed:
add vumi conversation key

sewagodimo committed:
remove the space in pytest config

sewagodimo committed:
remove the onniepie settings from base.py

sewagodimo committed:
remove the compress command

sewagodimo committed:
fix the secret key settings

miltontony commented on .travis.yml:
```diff
@@ -1,35 +1,40 @@
+        - pip install seaworthy[pytest]
+      script:
+        - docker build -t malaria .
+        - pytest seaworthy_docker_test.py
```
```suggestion
        - SECRET_KEY=malaria-test-key pytest seaworthy_docker_test.py
```

miltontony committed:
Apply suggestions from code review

add the secret key setting

Co-Authored-By: sewagodimo <sewagodimo.matlapeng@gmail.com>

sewagodimo committed:
quick twick to fix the docker_settings check

the docker settings check the a secret key has been set in the
enviroment else it breaks. This quick fix will go over that for now

sewagodimo committed:
remove secret key setting from pytest command

sewagodimo committed:
fix file name of travis

sewagodimo committed:
move seaworthy files into their own directory

sewagodimo committed:
fix flake errors

sewagodimo committed:
move the seaworthy dir up

sewagodimo committed:
add tests.py to travis commandd

sewagodimo committed:
remove docker's database

this requires rerunning the migrations with python3 which will fail

sewagodimo committed:
remove flake8 error

sewagodimo committed:
add no migrations to travis command

sewagodimo committed:
move test commands to pytest.ini

miltontony commented on malaria24/settings/base.py:
```diff
@@ -25,7 +26,9 @@
-SECRET_KEY = "*y%(_-arf724&tohkw4foyh_4wc6yqe=a7m^wkmz+kw+#pzfx*"
+
+DEFAULT_SECRET_KEY = 'please-change-me'
+SECRET_KEY = environ.get('SECRET_KEY') or DEFAULT_SECRET_KEY + "-"
```
This is odd. Why do you need `+ "-"` for?

sewagodimo replied:
the docker settings check that the secret key and default secret key is not the same, since we don't have the secret key settings in travis yet this was a quick fix to make the tests work

sewagodimo replied:
I was gonna remove it once the secret key was actually set

miltontony replied:
👍

jerith replied:
Probably better to set the `SECRET_KEY` envvar for the docker tests. :-)

sewagodimo committed:
comment out travis seaworthy tests

sewagodimo committed:
remove docker databse settings

sewagodimo committed:
add a seaworthy fixture for postgres

sewagodimo committed:
remove unused import

sewagodimo committed:
add the ve and ve3 to the dockerignore

sewagodimo committed:
uncomment the seaworthy tests

sewagodimo committed:
add postgres to the container

sewagodimo committed:
remove settings from pytest.ini

sewagodimo committed:
rename the seawothy fixtures file

sewagodimo committed:
update the seaworthy tests to to use fixtures

sewagodimo committed:
remove the secret key quick fix

sewagodimo committed:
set up a docker secret key

sewagodimo committed:
fix flake8 error

JayH5 commented on Dockerfile:
```diff
@@ -0,0 +1,18 @@
+# This is a development Dockerfile. For versioned Dockerfiles see:
+# https://github.com/praekeltfoundation/docker-seed
```
I think these comments are out-of-date?

sewagodimo replied:
removed them

JayH5 commented on Dockerfile:
```diff
@@ -0,0 +1,18 @@
+# This is a development Dockerfile. For versioned Dockerfiles see:
+# https://github.com/praekeltfoundation/docker-seed
+FROM praekeltfoundation/django-bootstrap:py2.7-jessie
```
Do you need to use `jessie`? If things work without the `-jessie` I'd prefer that as Debian Jessie is kind of out-of-date these days.

sewagodimo replied:
it seems like jessie is the most recent one on this list: https://cloud.docker.com/u/praekeltfoundation/repository/docker/praekeltfoundation/django-bootstrap

JayH5 replied:
We still build Jessie images because they're needed in some cases. But unless you need a specific Debian version I would recommend without the `-jessie` which gives you the latest version (which right now is `stretch`).

jerith replied:
Probably better to use the least specific base image you can. If it turns out an update to stretch or something breaks stuff, we can decide whether to fix whatever's breaking or explicitly build on jessie.

sewagodimo replied:
Oki, I removed the version pin

JayH5 commented on malaria24/settings/docker.py:
```diff
@@ -0,0 +1,31 @@
+from .production import *  # noqa: F401, F403
+
+
+SECRET_KEY = environ.get('DOCKER_SECRET_KEY') or DEFAULT_SECRET_KEY
```
Is there a reason this is called `DOCKER_SECRET_KEY`? Not sure if any of our other apps do this. Also if someone accidentally sets the `SECRET_KEY` env var when using Docker, they'll just end up with the default key.

sewagodimo replied:
yeah, I change it back to the secret key

JayH5 commented on requirements.txt:
```diff
@@ -1,4 +1,6 @@
 django-compressor<1.6
+django-modelcluster==1.1
+django-taggit==0.18.0
```
Where are these new requirements coming from?

sewagodimo replied:
those packages are pulling the latest version(which breaks) when they come in as dependencies.

jerith replied:
The Travis config installs `django-modelcluster==3.1` -- do we want `1.1` instead?

sewagodimo replied:
I fixed it

miltontony reviewed (approved):
👍 🍰 
Thanks for seeing this through!

JayH5 commented on seaworthy/fixtures.py:
```diff
@@ -0,0 +1,42 @@
+            }
+        }
+
+    def exec_django_admin(self, *args):
```
Doesn't look like you're using this method?

sewagodimo replied:
well remove it

JayH5 commented on seaworthy/fixtures.py:
```diff
@@ -0,0 +1,42 @@
+    "clean_postgresql_fixture",
+    "django_fixture",
+    "postgresql_fixture",
+]
```
Trailing newline.

sewagodimo replied:
added

JayH5 commented on seaworthy/fixtures.py:
```diff
@@ -0,0 +1,42 @@
+    WAIT_PATTERNS = (r"Booting worker",)
+
+    def __init__(self, name, db_url,
+                 image="malaria"):
```
Think this `__init__()` can be on one line?

sewagodimo replied:
yes it can

JayH5 reviewed (commented):
Looks good, couple of things to clean up.

sewagodimo committed:
rename setting

sewagodimo committed:
formatting fixes

sewagodimo committed:
remove old dockerfile comments

sewagodimo committed:
remove python-jessie version pin in the docerfile

sewagodimo committed:
remove unrequired version pins

sewagodimo committed:
add django-modelcluster to requirements.py

JayH5 reviewed (approved):
👍 🍰 looks good to me
